nomos-services/data-availability/network/src/backends/libp2p/validator.rs

```diff
@@ -68,15 +69,18 @@ pub enum DaNetworkEvent {
 /// Internally uses a libp2p swarm composed of the [`ValidatorBehaviour`]
 /// It forwards network messages to the corresponding subscription
 /// channels/streams
-pub struct DaNetworkValidatorBackend<Membership> {
+pub struct DaNetworkValidatorBackend<Membership>
+where
+    Membership: MembershipHandler,
+{
     task: (AbortHandle, JoinHandle<Result<(), Aborted>>),
     replies_task: (AbortHandle, JoinHandle<Result<(), Aborted>>),
     sampling_request_channel: UnboundedSender<(SubnetworkId, BlobId)>,
     balancer_command_sender: UnboundedSender<ConnectionBalancerCommand<BalancerStats>>,
     monitor_command_sender: UnboundedSender<ConnectionMonitorCommand<MonitorStats>>,
     sampling_broadcast_receiver: broadcast::Receiver<SamplingEvent>,
     verifying_broadcast_receiver: broadcast::Receiver<DaShare>,
-    _membership: PhantomData<Membership>,
+    membership: Arc<SwappableMembershipHandler<Membership>>,
```

Comment:
If `SwappableMembershipHandler` is an `Arc` already, why are we wrapping in `Arc` again?